src/components/web3/SendCryptoPopup.tsx

```diff
@@ -0,0 +1,663 @@
+  justify-content: center;
+  align-items: center;
+  flex: 1 0 0;
+`;
+
+const popupHeaderCSS = css`
+  display: flex;
+  height: 56px;
+  padding: 8px 16px 8px 24px;
+  align-items: center;
+  gap: 16px;
+  align-self: stretch;
+  font-family: Poppins;
+  font-size: 16px;
+  font-style: normal;
+  font-weight: 600;
+  line-height: 24px; /* 150% */
+`;
+
+const inputCSS = css`
+  width: 80px;
+  height: 16px;
+  gap: 8px;
+  border: 0px;
+  font-size: 16px;
+  border-bottom: 1px solid #a1abba;
+  font-style: normal;
+  font-weight: 400;
+  line-height: 24px; /* 171.429% */
+  vertical-align: middle;
+  &:focus {
+    border-bottom: 2px solid #3f39e8;
+  }
+  outline: none;
+`;
+
+const highlightBoxCSS = css`
+  border-radius: 4px;
+  background: #e9eeff;
+  padding: 16px;
+`;
+
+const Divider = () => {
+  return (
+    <svg
+      xmlns="http://www.w3.org/2000/svg"
+      width="500px"
+      height="1"
+      viewBox="0 0 500 1"
+      fill="none"
+      css={css`
+        display: flex;
+        height: 1px;
+        justify-content: center;
+        align-items: center;
+        align-self: stretch;
+      `}
+    >
+      <rect width="500" height="1" fill="#A1ABBA" fill-opacity="0.4" />
+    </svg>
+  );
+};
+
+const getNameFromId = async (jitsi: IJitsiMeetApi, id: string) => {
+  const info = (await jitsi.getRoomsInfo()).rooms;
+  const room = info.filter((r: any) => r.isMainRoom)[0];
+  const name = room.participants.filter((p: any) => p.id === id)[0].displayName;
+
+  return name;
+};
+
+export interface CryptoTransactionParams {
+  senderDisplayName: string;
+  sender: string;
+  recipient: string;
+  recipientDisplayName: string;
+  amount: number;
+  token: AllowedERC20Tokens;
+  nonce: string;
+}
+
+export interface SIWEReturnParams {
+  proof: {
+    signer: string;
+    signature: string;
+    payload: string;
+  };
+  method: string;
+}
+
+interface TransactionPendingResolution {
+  senderDisplayName: string;
+  siwe: SIWEReturnParams;
+}
+
+interface CryptoSendMessage {
+  type: "crypto";
+  msgType: "REQ" | "REJECT" | "SIGNED";
+  payload: SIWEReturnParams | CryptoTransactionParams | string;
+}
+
+interface CryptoPopupProps {
+  incomingRequests: CryptoTransactionParams[];
+  setIncomingRequests: React.Dispatch<
+    React.SetStateAction<CryptoTransactionParams[]>
+  >;
+  web3Address: string;
+  jitsi: IJitsiMeetApi;
+}
+
+const DEBUG_SHOW = true;
+
+export const CryptoRecievePopup: React.FC<CryptoPopupProps> = ({
+  incomingRequests,
+  setIncomingRequests,
+  web3Address,
+  jitsi,
+}) => {
+  const [showing, setShowing] = useState(DEBUG_SHOW);
+  const [params, setParams] = useState({} as any);
+
+  const returnSIWEToOrigin = async () => {
+    let siwe: SIWEReturnParams;
+    try {
+      siwe = await generateSIWEForCrypto(
+        web3Address,
+        params,
+        `Please sign this message so that ${params.senderDisplayName} knows that you own this address, and may send you ${params.amount} ${params.token}.`
```

Comment:
We should apply simple validation to amount and token (not sure if here or earlier in message processing) and reject message if amount not an int(?) or token not in enum rather than displaying arbitrary text. Best place is probably somewhere before `setParams(incomingRequests[0]);` below.